### ISSUE TITLE
Incorrect code generated for Bytes.get_int64_le on ppc64

### ISSUE THREAD

**Issue by dra27** [ocaml/ocaml#10302]
```ocaml
let _ = Bytes.(get_int64_le (unsafe_of_string "\xfe\xdc\xba\x98\x76\x54\x32\x10\x01\x00") 1)
```
on ppc64 generates:
```asm
        ld      21, 1(11)
```
which gives assembler error "operand out of domain (1 is not a multiple of 4)".

**Comment by xavierleroy** [comment]
Well spotted.  We need to be more careful during instruction selection and not select the `ld OFFSET(reg)` and `sd OFFSET(reg)` forms unless OFFSET is a multiple of 4.

**Comment by bschommer** [comment]
Seems to me that the gnu assembler here might be more strict, from the PowerPC ISA Document I can get the following restriction for load/stores with DS-FORM:

> The DS field in DS-form Storage Access instruc-
tions is a word offset, not a byte offset like the D
field in D-form Storage Access instructions. How-
ever, for programming convenience, Assemblers
should support the specification of byte offsets for
both forms of instruction.

**Comment by xavierleroy** [comment]
I read this documentation as "in assembly language, the offset is given in bytes, but it must be a multiple of 4 because in the opcode it is stored as a word offset".  So, the GNU assembler is doing the right thing by rejecting `ld 21, 1(11)`.

**Comment by bschommer** [comment]
> I read this documentation as "in assembly language, the offset is given in bytes, but it must be a multiple of 4 because in the opcode it is stored as a word offset". So, the GNU assembler is doing the right thing by rejecting `ld 21, 1(11)`.

Make sense, that is by the way the only place were I could find any restriction in the ISA Manual.

**Comment by Octachron** [comment]
Fixed by #10303 .
